### ISSUE TITLE
Problems with getting stuck during training

### ISSUE THREAD

**Issue by qweasd982** [m-abr/FCPCodebase#54]
I encountered a training freeze. Around 10 million time steps, the terminal stopped outputting training-related data (number of epochs, learning rate, loss), stopped training, and stopped without reporting an error. When I used the Roboviz visualization software to view it, the screen was black. I had to terminate training and then resume it (restrain).

**Comment by m-abr** [comment]
The most likely scenario is that the gym code got stuck in some loop. The black screen in Roboviz means that it's not getting new data from the agent, which usually means the agent is stuck, most likely due to an issue with the gym environment, and not the RL algorithm. The error might be hard to catch since it's rare, but I’d suggest focusing on your gym environment and adding detailed logging to help identify where the problem is.

**Comment by qweasd982** [comment]
<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/ffc39819-6349-4d02-9e7d-119cc95de0aa" />
<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/b618ddd0-d6b0-4718-bd5b-c23f83d07641" />
In the first picture, Roboviz was already black at that time, but the training was still going on. The CPU usage was 100% at this time. But after a while, the training seemed to be stuck, the terminal no longer continued to output information, and the CPU usage was very low.

**Comment by qweasd982** [comment]
> 最可能的情况是 Gym 代码卡在了某个循环中。Roboviz 的黑屏意味着它没有从代理获取新数据，这通常意味着代理卡住了，很可能是由于 Gym 环境的问题，而不是强化学习算法的问题。由于这种错误很少见，可能很难发现，但我建议你关注 Gym 环境，并添加详细的日志记录，以帮助识别问题所在。

After the roboviz screen goes black, you can continue training for a while.

**Comment by m-abr** [comment]
I haven’t encountered that issue before. Is it possible that your gym environment is still executing the step() function but not actually sending actions to the server? That could explain why training continues even after Roboviz goes black. In theory, this could happen even if reset() works fine (if an episode can run indefinitely or for a very large number of steps, keeping the agent stuck in a faulty step() loop). Also, can you check whether Roboviz is black for all the other training processes too, or just this one? 
In any case it would help to log what’s happening inside your gym environment to understand better what’s happening.

**Comment by qweasd982** [comment]
> 我以前没遇到过这个问题。你的 Gym 环境是否可能仍在执行 step() 函数，但实际上并没有向服务器发送操作？这或许可以解释为什么 Roboviz 断电后训练仍在继续。理论上，即使 reset() 函数运行正常，这种情况也可能会发生（如果一个回合可以无限期运行，或者运行大量步骤，导致代理陷入错误的 step() 循环）。另外，你能检查一下 Roboviz 是否在所有其他训练过程中都断电，还是只有这一次断电？ 无论如何，记录 Gym 环境中发生的事情有助于更好地了解正在发生的事情。

Thank you very much for your help
